src/Microsoft.CSharp/tests/ArrayHandling.cs

```diff
@@ -31,25 +32,31 @@ public void ArrayTypeNames()
             d = new int[3];
             ex = Assert.Throws<RuntimeBinderException>(() => { string s = d; });
             Assert.Contains("int[]", ex.Message);
+        }
 
-            d = new int[3, 2, 1];
-            ex = Assert.Throws<RuntimeBinderException>(() => { string s = d; });
+        [Fact]
+        [SkipOnTargetFramework(TargetFrameworkMonikers.UapAot, "https://github.com/dotnet/corert/issues/3331")]
+        public void MultiDimArrayTypeNames()
+        {
+            dynamic d = new int[3, 2, 1];
+            RuntimeBinderException ex = Assert.Throws<RuntimeBinderException>(() => { string s = d; });
             Assert.Contains("int[,,]", ex.Message);
```

Comment:
The first half of this test should work fine - it's using zero bounds. Better to the split the test and disable only the second half. Also, 3331 is about rank-1 multidims, not NZ bounds. The test you want to use is PlatformDetection.IsNonZeroLowerBoundArraySupported.